packages/bruno-electron/src/ipc/network/authorize-user-in-window.js

```diff
@@ -30,12 +30,12 @@ const authorizeUserInWindow = ({ authorizeUrl, callbackUrl, session }) => {
     });
 
     function onWindowRedirect(url) {
-      // check if the url contains an authorization code
-      if (new URL(url).searchParams.has('code')) {
-        finalUrl = url;
-        if (!url || !finalUrl.includes(callbackUrl)) {
-          reject(new Error('Invalid Callback Url'));
+      // check if the redirect is to the callback URL and if it contains an authorization code
+      if (url && url.includes(callbackUrl)) {
```

Comment:
I noticed a case where this check may not be sufficient (which is the problem in this version as it was in the original):
My keycloak server has a redirect uri set to exactly: `https://example.com?param=value`. As keycloak expects me to use callback url exactly as configured, I do so in bruno, but after I provide correct credentials, I get redirected to `https://example.com/?param=value&code=...`, whis does not pass the check `url.includes(callbackUrl)` (extra `/` breaks this).  
Perhaps the string comparison using `includes` should be done on `URL.href`'s rather than plain string values, to have a level of normalization?
```
let strurl1 = 'https://example.com?par=val'
let strurl2 = 'https://example.com/?par=val'
strurl1 == strurl2 // ==>false
new URL(strurl1) == new URL(strurl2); // ==> false
new URL(strurl1).href == new URL(strurl2).href; // ==> true
new URL(strurl1).href // => https://example.com/?par=val 
new URL(strurl2).href // => https://example.com/?par=val 
```

Reply:
@helloanoop @lohit-1 Do we have any unit tests on this behaviour? I was not able to find any when creating the PR but it seems like there are several edge cases here so I would like to add some if they are not there

Reply:
@lohit-1 Have added some test cases, please review.

I couldn't write a test for the entire component, including mocking all the `window` events, so I just extracted that check into a separate function and wrote tests for it.